docs/install.md

```diff
@@ -80,30 +80,51 @@ As a summary, after installing Magisk in recovery **(starting from power off)**:
 
-Unlocking the bootloader on modern Samsung devices have some caveats. The newly introduced `VaultKeeper` service will make the bootloader reject any unofficial partitions in some circumstances.
+### Requirements
+
+To verify whether or not Magisk can be installed in your Samsung device, you first must check the OEM Lock and KnoxGuard (RMM) status, to do so boot your device in Download mode with its key combo.
+
+Possible OEM Lock values are the following:
+- **ON (L)**: fully locked.
+- **ON (U)**: bootloader locked, OEM unlocking enabled.
+- **OFF (U)**: fully unlocked.
+
+To unlock your bootloader, follow the instructions below. If no OEM Lock value is shown in Download mode, your device is probably not unlockable due to market limitations (USA/Canada devices).
+
+Possible KnoxGuard values are the following:
+
+- `Active`, `Locked`: your device has been remotely locked by your telecom operator or your insurance company.
+- `Prenormal`: your device is temporarely locked, reaching 168h of uptime should trigger unlock.
+- `Checking`, `Completed`, `Broken`: your device is unlocked.
+
+Having KnoxGuard active will prevent you from installing/running Magisk regardless of your bootloader lock state.
+
+### Unlocking the bootloader
 
 - Allow bootloader unlocking in **Developer options → OEM unlocking**
 - Reboot to download mode: power off your device and press the download mode key combo for your device
 - Long press volume up to unlock the bootloader. **This will wipe your data and automatically reboot.**
-- Go through the initial setup. Skip through all the steps since data will be wiped again in later steps. **Connect the device to Internet during the setup.**
-- Enable developer options, and **confirm that the OEM unlocking option exists and is grayed out.** This means the `VaultKeeper` service has unleashed the bootloader.
-- Your bootloader now accepts unofficial images in download mode
 
 ### Instructions
 
-- Use either [samfirm.js](https://github.com/jesec/samfirm.js), [Frija](https://forum.xda-developers.com/s10-plus/how-to/tool-frija-samsung-firmware-downloader-t3910594), or [Samloader](https://forum.xda-developers.com/s10-plus/how-to/tool-samloader-samfirm-frija-replacement-t4105929) to download the latest firmware zip of your device directly from Samsung servers.
+- Download the latest firmware package for your device, you can use one of the tools below to download it directly from Samsung servers:
+  - [SamFirm.NET](https://github.com/jesec/SamFirm.NET), [samfirm.js](https://github.com/jesec/samfirm.js)
+  - [Frija](https://forum.xda-developers.com/s10-plus/how-to/tool-frija-samsung-firmware-downloader-t3910594)
+  - [Samloader](https://forum.xda-developers.com/s10-plus/how-to/tool-samloader-samfirm-frija-replacement-t4105929)
+  - [Bifrost](https://forum.xda-developers.com/t/tool-samsung-samsung-firmware-downloader.4240719/)
```

Comment:
```suggestion
  - [Bifrost](https://forum.xda-developers.com/t/tool-samsung-samsung-firmware-downloader.4240719/)
  - [Samloader kotlin](https://github.com/zacharee/SamloaderKotlin)
```

Reply:
Samloader Kotlin is already there, it's actually called "Bifrost":

https://github.com/zacharee/SamloaderKotlin/blob/64fc960a3c7d0eb491bd465db564a7eeebed5b2a/README.md?plain=1#L6

But I can replace the XDA thread link with the GitHub repo page if you want.